filebeat/input/v2/input.go

```diff
@@ -79,6 +81,12 @@ type Context struct {
 	// The input ID.
 	ID string
 
+	// GeneratedID is true when the config does not include an ID and therefore
+	// the runner uses a hash from the config as ID. It's used to know if the
+	// v2.Context generated for the input should have a non-discard metric
+	// registry.
+	GeneratedID bool
```

Comment:
If the whole point of this field is to know if the `v2.Context` should have a valid/published metrics registry, why not give it a more meaningful name? Maybe something like `PublishMetrics`, or `MetricsEnabled`.

Reply:
done: https://github.com/elastic/beats/pull/42618/commits/da345c0f59c295112598014dd572d3a5a69f6819